docs/control-structures.rst

```diff
@@ -51,40 +54,63 @@ write::
 
 The names of output parameters can be omitted.
 The output values can also be specified using ``return`` statements.
-The ``return`` statements are also capable of returning multiple
-values, see :ref:`multi-return`.
+
+The ``return`` statements are also capable of returning multiple values using ``return (v0, v1, ..., vn)``. The number of parameters must equal the number of output parameters.
+
 Return parameters are initialized to zero; if they are not explicitly
 set, they stay to be zero.
 
 Input parameters and output parameters can be used as expressions in
 the function body.  There, they are also usable in the left-hand side
 of assignment.
 
+.. index:: return array, return string, array, string, array of strings, dynamic array, variable sized array
+
+.. _return-array:
+
+Returning Strings and Arrays
+----------------------------
+
+You can return strings and arrays from Solidity functions, see `array_receiver_and_returner.sol <https://github.com/fivedogit/solidity-baby-steps/blob/master/contracts/60_array_receiver_and_returner.sol>`_ for an example.
+
+.. index:: return struct, struct
+
+.. _return-struct:
+
+Returning Structs
+-----------------
+
+You can return a ``struct`` from a Solidity function, but only from ``internal`` function calls.
+
 .. index:: if, else, while, do/while, for, break, continue, return, switch, goto
 
 Control Structures
 ===================
 
-Most of the control structures from JavaScript are available in Solidity
-except for ``switch`` and ``goto``. So
-there is: ``if``, ``else``, ``while``, ``do``, ``for``, ``break``, ``continue``, ``return``, ``? :``, with
-the usual semantics known from C or JavaScript.
+Supported Expressions
+---------------------
 
-Parentheses can *not* be omitted for conditionals, but curly brances can be omitted
-around single-statement bodies.
+Most of the `control structures from JavaScript <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Control_flow_and_error_handling>`_ are available in Solidity except for ``switch`` and ``goto``. The full list is:
 
-Note that there is no type conversion from non-boolean to boolean types as
-there is in C and JavaScript, so ``if (1) { ... }`` is *not* valid
-Solidity.
+- ``if``
```

Comment:
Perhaps better to list them like this:

 - `if (<condition>) <true-statement>`
 - `if (<condition>) <true-statement> else <false-statement>`

and so on

Reply:
Done